termwiz/src/lineedit/buffer.rs

```diff
@@ -0,0 +1,189 @@
+use unicode_segmentation::GraphemeCursor;
+
+use super::actions::Movement;
+
+pub struct LineEditBuffer {
+    line: String,
+    /// byte index into the UTF-8 string data of the insertion
+    /// point.  This is NOT the number of graphemes!
+    cursor: usize,
+}
+
+impl Default for LineEditBuffer {
+    fn default() -> Self {
+        Self {
+            line: String::new(),
+            cursor: 0,
+        }
+    }
+}
+
+impl LineEditBuffer {
+    pub fn new(line: &str, cursor: usize) -> Self {
+        let mut buffer = Self::default();
+        buffer.set_line_and_cursor(line, cursor);
+        return buffer;
+    }
+
+    pub fn get_line(&self) -> &String {
```

Comment:
Since this is immutable, it is considered "better" to return `&str` rather than `&String`:

```suggestion
    pub fn get_line(&self) -> &str {
```

Reply:
Fixed